collect_app/src/main/java/org/odk/collect/android/widgets/utilities/RecordingStatusProvider.java

```diff
@@ -0,0 +1,14 @@
+package org.odk.collect.android.widgets.utilities;
+
+import android.util.Pair;
+
+import org.javarosa.form.api.FormEntryPrompt;
+
+import java.util.function.Consumer;
+
+public interface RecordingStatusProvider {
+
+    void onIsRecordingBlocked(Consumer<Boolean> isRecordingListener);
```

Comment:
Is this idiomatic naming? That is, these methods don't feel like callbacks/handlers, they're setting up observers. I know this code was already there previously but it really jumped out at me this time, maybe because of the `Provider` naming. This doesn't feel like a provider. It feels almost like `RecordingStatusObserver` with `observeRecordingBlocked` and `observeRecordingInProgress`. Not perfect because `Observer` is a pattern but I'm bringing this up because my first reaction is that I didn't like the structure. I now think it's actually ok, I just can't reconcile it with the naming. Perhaps it's just an idiom I'm not familiar with.

Reply:
Maybe `RecordingStatusHandler` and `onBlockedStatusChange` and `onRecordingStatusChange`?

I'm not sure this is even really worth discussing, I'm probably just not writing enough code and also getting old. 😄

Reply:
No I also felt weird about the naming once I broke it out but thought that might just be me.  I like your last suggestion, so I'll go with that for the moment, and we can see how it settles.